fe/fe-core/src/main/java/org/apache/doris/nereids/properties/RequestPropertyDeriver.java

```diff
@@ -71,21 +71,21 @@ public Void visit(Plan plan, PlanContext context) {
 
     @Override
     public Void visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> hashJoin, PlanContext context) {
-        // for broadcast join
-        List<PhysicalProperties> propertiesForBroadcast = Lists.newArrayList(
-                new PhysicalProperties(),
-                new PhysicalProperties(new DistributionSpecReplicated())
-        );
         // for shuffle join
-        Pair<List<SlotReference>, List<SlotReference>> onClauseUsedSlots = JoinUtils.getOnClauseUsedSlots(hashJoin);
-        List<PhysicalProperties> propertiesForShuffle = Lists.newArrayList(
-                new PhysicalProperties(new DistributionSpecHash(onClauseUsedSlots.first, ShuffleType.JOIN)),
-                new PhysicalProperties(new DistributionSpecHash(onClauseUsedSlots.second, ShuffleType.JOIN)));
-
         if (!JoinUtils.onlyBroadcast(hashJoin)) {
+            Pair<List<SlotReference>, List<SlotReference>> onClauseUsedSlots = JoinUtils.getOnClauseUsedSlots(hashJoin);
+            List<PhysicalProperties> propertiesForShuffle = Lists.newArrayList(
+                    new PhysicalProperties(new DistributionSpecHash(onClauseUsedSlots.first, ShuffleType.JOIN)),
+                    new PhysicalProperties(new DistributionSpecHash(onClauseUsedSlots.second, ShuffleType.JOIN)));
+
             requestPropertyToChildren.add(propertiesForShuffle);
         }
+        // for broadcast join
         if (!JoinUtils.onlyShuffle(hashJoin)) {
+            List<PhysicalProperties> propertiesForBroadcast = Lists.newArrayList(
+                    new PhysicalProperties(),
```

Comment:
i think it is better to create any distribution properties explicitly. such as
```
new PhysicalProperties(DistributionSpecAny.getInstance())
```
or
```
PhysicalProperties.CreateDistributionSpecAnyInstance();
```